replica/database.hh

```diff
@@ -402,19 +403,17 @@ private:
     uint64_t _failed_counter_applies_to_memtable = 0;
 
     template<typename... Args>
-    void do_apply(db::rp_handle&&, Args&&... args);
-
-    lw_shared_ptr<memtable_list> _memtables;
+    void do_apply(compaction_group& cg, db::rp_handle&&, Args&&... args);
 
     lw_shared_ptr<memtable_list> make_memory_only_memtable_list();
-    lw_shared_ptr<memtable_list> make_memtable_list();
+    lw_shared_ptr<memtable_list> make_memtable_list(compaction_group& cg);
 
+    compaction_manager& _compaction_manager;
     sstables::compaction_strategy _compaction_strategy;
-    // SSTable set which contains all non-maintenance sstables
-    lw_shared_ptr<sstables::sstable_set> _main_sstables;
-    // Holds SSTables created by maintenance operations, which need reshaping before integration into the main set
-    lw_shared_ptr<sstables::sstable_set> _maintenance_sstables;
-    // Compound set which manages all the SSTable sets (e.g. main, etc) and allow their operations to be combined
+    // TODO: Still holds a single compaction group, meaning all sstables are eligible to be compacted with one another. Soon, a table
+    //  will be able to hold more than one group.
```

Comment:
How about keeping a map of compaction_group per token range
and populate it with a single entry for the full range?

This will show the basic requirements of supporting multiple groups upfront
and will reduce the churn of supporting multiple groups later on.

Reply:
I think it's okay to start with unique_ptr. Since there's just one now all the code to locate the correct group will be buggy.